client/audio/audio_qt.cpp

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: GPLv3-or-later
+// SPDX-FileCopyrightText: Freeciv authors
+// SPDX-FileCopyrightText: Freeciv21 authors
+// SPDX-FileCopyrightText: Louis Moureaux <m_louis30@yahoo.com>
+
+#include "audio_qt.h"
+
+// utility
+#include "log.h"
+
+// client
+#include "audio.h"
+
+// Qt
+#include <QAudio>
+#include <QAudioOutput>
+#include <QMediaPlayer>
+#include <QSoundEffect>
+
+namespace {
+static QMediaPlayer *player = nullptr;
+static QAudioOutput *output = nullptr;
+static double audio_volume = 1.0;
+audio_finished_callback current_callback;
+static std::map<QString, QSoundEffect> cache = {};
+} // namespace
+
+/**
+ * Set the volume.
+ */
+static void qt_audio_set_volume(double volume)
+{
+  audio_volume = QAudio::convertVolume(
+      volume, QAudio::LogarithmicVolumeScale, QAudio::LinearVolumeScale);
+  output->setVolume(audio_volume);
+}
+
+/**
+ * Get the volume.
+ */
+static double qt_audio_get_volume()
+{
+  return QAudio::convertVolume(audio_volume, QAudio::LinearVolumeScale,
+                               QAudio::LogarithmicVolumeScale);
+}
+
+/**
+ * Play sound
+ */
+static bool qt_audio_play(const QString &tag, const QString &fullpath,
+                          bool repeat, audio_finished_callback cb)
+{
+  if (fullpath.isEmpty()) {
+    return false;
+  }
+
+  if (repeat) {
+    // Repeating media is used for music. Since latency isn't critical here,
+    // we use QMediaPlayer.
+    player->setSource(QUrl::fromLocalFile(fullpath));
+
+    if (!cb) {
+      player->setLoops(QMediaPlayer::Infinite);
+    }
+    current_callback = cb;
+
+    player->play();
+
+    qDebug() << "Playing music file" << fullpath;
+
+  } else {
+    // Non-repeating media is used for sound effects. Latency is more
+    // important for this, so we use QSoundEffect.
+    if (cache.count(fullpath) == 0) {
```

Comment:
How much is the latency different between the `QSoundEffect` an the `QMediaPlayer`? The move to WAV files seems problematic.

Reply:
WAV files add 1MB to the repo and installer, which is significant but not too much.